File: TDS/TDS/Models/Usuario.cs
using System;
using System.Collections.Generic;

namespace TDS.Models
{
    public partial class Usuario
    {
        public int Id { get; set; }
        public string? Email { get; set; }
        public string? Password { get; set; }
        public int? EstudianteId { get; set; }
        public int? MaestroId { get; set; }
        public int? RolId { get; set; }
        public int? InstitucionId { get; set; }
        public bool? Estado { get; set; }

        public virtual Estudiante? Estudiante { get; set; }
        public virtual Institucion? Institucion { get; set; }
        public virtual Maestro? Maestro { get; set; }
        public virtual Rol? Rol { get; set; }
    }
}

File: TDS_Front/src/app/models/usuario.ts
import { Estudiante } from "./estudiante";
import { Maestro } from "./maestro";

export class Usuario {
  constructor(public id: number = 0, public password: string = "", public estudianteId: number | null = 0, public maestroId: number | null = 0, public rolId: number = 0, public institucionId: number = 0, public estado: boolean = false, public estudiante: Estudiante | null = new Estudiante(), public maestro: Maestro | null = new Maestro()){}
}
